substrate/client/executor/polkavm/src/lib.rs

```diff
@@ -58,22 +58,23 @@ impl WasmInstance for Instance {
 		};
 
 		// TODO: This will leak guest memory; find a better solution.
-		let mut state_args = polkavm::StateArgs::new();
 
 		// Make sure the memory is cleared...
-		state_args.reset_memory(true);
-		// ...and allocate space for the input payload.
-		state_args.sbrk(raw_data_length);
+		if self.0.reset_memory().is_err() {
+			return (
+				Err(format!("call into the runtime method '{name}' failed: initialize memory")
```

Comment:
The error message from `reset_memory` is still not printed out. Should be something like:

`Err(format!("call into the runtime method '{name}' failed: reset memory failed: {error}")`